pkg/koordlet/statesinformer/states_gpu.go

```diff
@@ -0,0 +1,207 @@
+/*
+Copyright 2022 The Koordinator Authors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package statesinformer
+
+import (
+	"context"
+	"os"
+	"time"
+
+	"github.com/NVIDIA/go-nvml/pkg/nvml"
+	corev1 "k8s.io/api/core/v1"
+	"k8s.io/apimachinery/pkg/api/errors"
+	"k8s.io/apimachinery/pkg/api/resource"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/klog/v2"
+
+	"github.com/koordinator-sh/koordinator/apis/extension"
+	schedulingv1alpha1 "github.com/koordinator-sh/koordinator/apis/scheduling/v1alpha1"
+	"github.com/koordinator-sh/koordinator/pkg/koordlet/metriccache"
+)
+
+func generateQueryParam() *metriccache.QueryParam {
+	end := time.Now()
+	start := end.Add(-time.Duration(60) * time.Second)
+	return &metriccache.QueryParam{
+		Aggregate: metriccache.AggregationTypeLast,
+		Start:     &start,
+		End:       &end,
+	}
+}
+
+func (s *statesInformer) reportGPUDevice() {
+	s.nodeRWMutex.RLock()
+	copyNode := s.node.DeepCopy()
+	s.nodeRWMutex.RUnlock()
+	queryParam := generateQueryParam()
+	nodeResource := s.metricsCache.GetNodeResourceMetric(queryParam)
+	if nodeResource.Error != nil {
+		klog.Errorf("failed to get node resource metric, err: %v", nodeResource.Error)
+		return
+	}
+	if len(nodeResource.Metric.GPUs) == 0 {
+		klog.V(5).Info("no gpu device found")
+		return
+	}
+	var deviceInfos []schedulingv1alpha1.DeviceInfo
```

Comment:
consider for fpga\rdma's extensibility in the future,  we should not bind the gpu device crd generation process and update\create logic together.   I think the below process is better:

1. var deviceInfos []schedulingv1alpha1.DeviceInfo
2. fill gpu\fpga\rdma to deviceInfos
3. create\update remote crd